File: api/internal/models/contact_provider.go
package models

import (
	"time"

	"github.com/google/uuid"
)

const (
	SendgridContactProvider ContactProviderType = "sendgrid"
	BrevoContactProvider    ContactProviderType = "brevo"
)

type (
	ContactProviderType string
	ContactProvider     struct {
		ID     uuid.UUID `json:"id"`
		UserID uuid.UUID `json:"userId"`

		Type            ContactProviderType `json:"type"`
		APIKey          *string             `json:"apiKey,omitempty"`
		LastAPIKeyChars *string             `json:"lastApiKeyChars,omitempty"`

		CreatedAt time.Time  `json:"createdAt"`
		UpdatedAt *time.Time `json:"updatedAt,omitempty"`
	}
)

File: frontend/app/types/contactProvider.ts
export enum ContactProviderType {
    Salesforce = "salesforce",
    Hubspot = "hubspot",
    Zoho = "zoho",
    Sendgrid = "sendgrid",
    Mailchimp = 'mailchimp',
    Brevo = "brevo"
}

export interface ContactProvider {
    type: ContactProviderType
    latestApiKeyChars: string
}